pkg/http/server.go

```diff
@@ -123,6 +123,10 @@ func (srv *Server) Shutdown() error {
 		return errors.New("http server already in shutdown")
 	}
 
+	if srv.listener == nil {
+		return nil
```

Comment:
A racy Start() and Stop() would set `srv.inShutdown` incremented.  It would look like the server is in shutdown.

You could move this check above `if atomic.AddUint32(...` block.